client/pages/edit-entry-form.jsx

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+
+class EditForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      entry: null,
+      categoryList: [],
+      amount: null,
+      date: null,
+      description: null,
+      categoryId: null
+    };
+
+    this.handleChangeCategory = this.handleChangeCategory.bind(this);
+    this.handleChangeDate = this.handleChangeDate.bind(this);
+    this.handleChangeAmount = this.handleChangeAmount.bind(this);
+    this.handleChangeDescription = this.handleChangeDescription.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  componentDidMount() {
+    fetch(`/api/entry/${this.props.entryId}`)
+      .then(response => response.json())
+      .then(entry => this.setState({ entry }));
+
+    fetch('/api/category-table')
+      .then(response => response.json())
+      .then(categoryList => this.setState({ categoryList }));
+  }
+
+  handleChangeCategory(event) {
+    const newEntry = this.state.entry;
+    newEntry.category = event.target.value;
+    this.setState({ entry: newEntry });
+  }
+
+  handleChangeDate() {
+    const newEntry = this.state.entry;
+    newEntry.date = event.target.value;
+    this.setState({ entry: newEntry });
+  }
+
+  handleChangeAmount() {
+    const newEntry = this.state.entry;
+    newEntry.amount = event.target.value;
+    this.setState({ entry: newEntry });
+  }
+
+  handleChangeDescription() {
+    const newEntry = this.state.entry;
+    newEntry.description = event.target.value;
+    this.setState({ entry: newEntry });
+  }
```

Comment:
These can be deduplicated with some bracket notation if you include a `name` prop on each input.
```js
handleChange(event) {
  const name = event.target.name;
  const value = event.target.value;
  const newState = {};
  newState[name] = value;
  this.setState(newState)
}
```